scalafmt-core/shared/src/main/scala/org/scalafmt/internal/FormatOps.scala

```diff
@@ -1767,23 +1773,32 @@ class FormatOps(
   }
 
   // Redundant () delims around case statements
-  def isCaseBodyEnclosedAsBlock(ft: FormatToken, caseStat: Case)(implicit
+  def isCaseBodyEnclosedAsBlock(ft: FormatToken, caseStatement: Tree)(implicit
       style: ScalafmtConfig
   ): Boolean = {
-    val body = caseStat.body
-    (ft.noBreak || style.newlines.getBeforeMultiline.ignoreSourceSplit) &&
-    body.eq(ft.meta.rightOwner) && (body match {
-      case _: Lit.Unit | _: Term.Tuple => false
-      case t: Term.ApplyInfix =>
-        val op = t.op.value
-        op != "->" && op != "→"
-      case _ => true
-    }) && {
-      val btoks = body.tokens
-      btoks.headOption.exists { head =>
-        head.is[Token.LeftParen] && matchingOpt(head).contains(btoks.last)
+
+    def isEnclosed(ft: FormatToken, caseBody: Tree)(implicit
+        style: ScalafmtConfig
+    ): Boolean = {
+      (ft.noBreak || style.newlines.getBeforeMultiline.ignoreSourceSplit) &&
+      caseBody.eq(ft.meta.rightOwner) && (caseBody match {
+        case _: Lit.Unit | _: Term.Tuple => false
+        case t: Term.ApplyInfix =>
+          val op = t.op.value
+          op != "->" && op != "→"
+        case _ => true
+      }) && {
+        val btoks = caseBody.tokens
+        btoks.headOption.exists { head =>
+          head.is[Token.LeftParen] && matchingOpt(head).contains(btoks.last)
+        }
       }
     }
+    val body = caseStatement match {
+      case tc: TypeCase => tc.body
+      case c: Case => c.body
```

Comment:
should we add an explicit default case here and throw?

Reply:
I added an option here.